x/delayedack/keeper/keeper.go

```diff
@@ -90,6 +90,18 @@ func (k Keeper) GetRollapp(ctx sdk.Context, chainID string) (rollapptypes.Rollap
 	return k.rollappKeeper.GetRollapp(ctx, chainID)
 }
 
+func (k Keeper) GetRollappFinalizedHeight(ctx sdk.Context, chainID string) (uint64, error) {
+	res, err := k.rollappKeeper.StateInfo(ctx, &rollapptypes.QueryGetStateInfoRequest{
+		RollappId: chainID,
+		Finalized: true,
+	})
+	if err != nil {
+		return 0, err
+	}
+
+	return (res.StateInfo.StartHeight + res.StateInfo.NumBlocks), nil
```

Comment:
should be here (res.StateInfo.StartHeight + res.StateInfo.NumBlocks -1) to get the actual height (missing minus 1)